sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1058,101 @@ def _eval_is_nonnegative(self):
+    ``expand_func()`` or ``expand(func=True)``. The former will keep
+    the polynomial in factored form while the latter will expand the
+    polynomial itself. See examples for details.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols, gamma
+    >>> n = Symbol('n', integer=True, positive=True)
+
+    >>> multinomial(15, 8)
+    490314
+
+    Comparing Coefficents generated with the multinomial function:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a+b+c)**3).expand(func=True)
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The Coefficient when of 2 1 0 corresponding to the second term is 3
+
+    >>> multinomial(2, 1, 0)
+    3
+
+    >>> multinomial(Rational(5, 4), 3)
+    gamma(21/4)/(6*gamma(9/4))
+
+    >>> multinomial(n, 3).rewrite(gamma)
+    gamma(n + 4)/(6*gamma(n + 1))
+
+    >>> multinomial(n, 3).expand(func=True)
+    n**3/6 + n**2 + 11*n/6 + 1
+
+    >>> expand_func(multinomial(n, 3))
+    (n + 1)*(n + 2)*(n + 3)/6
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multinomial_theorem#Multinomial_coefficients
+
+    .. [2] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/
+
+    """
+
+    @classmethod
+    def eval(self, *args):
+        args = list(map(sympify, args))
+        if len(args) < 2:
+            raise BadArgumentsError("A miniumum of 2 args required")
+        if any(x.is_Integer and x.is_negative for x in args):
+            return S.Zero
+        else:
+            if all(x.is_Integer for x in args):
+                from sympy import binomial
+                if len(args) == 2:
+                    return binomial(sum(args), args[1])
+                return Mul(*[binomial(j, k) for j, k in zip([Add(*args[:i]) for i in range(1, len(args) + 1)], args)])
+            elif all(x.is_number for x in args):
+                from sympy import gamma
+                return gamma(Add(*args) + 1)/Mul(*(gamma(x + 1) for x in args))
+
+    def _eval_rewrite_as_factorial(self, *args, **kwargs):
+        return factorial(Add(*args))/Mul(*(factorial(x) for x in args))
+
+    def _eval_expand_func(self, **hints):
+        from sympy import gamma, expand_func
+        return expand_func(gamma(Add(*self.args) + 1)/Mul(*(gamma(x + 1) for x in self.args)))
+
+    def _eval_rewrite_as_binomial(self, *args, **kwargs):
+        from sympy import binomial
+        return Mul(*[binomial(j, k) for j, k in zip([Add(*args[:i]) for i in range(1, len(args) + 1)], args)])
+
+    def _eval_rewrite_as_gamma(self, *k, **kwargs):
+        from sympy import gamma
+        return gamma(Add(*k) + 1)/Mul(*(gamma(x + 1) for x in k))
+
+    def _eval_is_integer(self):
+        k = list(map(sympify, self.args))
+        if all(x.is_integer for x in k):
+            return True
+        elif all(x.is_integer for x in k) is False:
```

Comment:
I don't think this is needed; if eval was able to evaluate it was because it knew whether items were integers or not. And if *it* couldn't decide, *this* won't be able to decide, either.

Reply:
```python
def _eval_is_integer(self):
    from sympy.core.logic import _fuzzy_group, fuzzy_or, fuzzy_and
    return fuzzy_or(
        _fuzzy_group(x.is_integer for x in self.args),
        any(fuzzy_and((x.is_integer, x.is_negative)) for x in self.args))
```

(btw, no need to sympify args of a SymPy object)

Reply:
gotcha